controllers/topology/reconcile_state.go

```diff
@@ -83,26 +83,24 @@ func (r *ClusterReconciler) reconcileControlPlane(ctx context.Context, s *scope.
 			current:              s.Current.ControlPlane.InfrastructureMachineTemplate,
 			desired:              s.Desired.ControlPlane.InfrastructureMachineTemplate,
 			compatibilityChecker: check.ReferencedObjectsAreCompatible,
-			templateNamer: func() string {
-				return controlPlaneInfrastructureMachineTemplateNamePrefix(s.Current.ControlPlane.Object.GetClusterName())
-			},
+			templateNamePrefix:   controlPlaneInfrastructureMachineTemplateNamePrefix(s.Current.ControlPlane.Object.GetClusterName()),
 		},
 		)
 		if err != nil {
-			return errors.Wrapf(err, "failed to update the %s object", s.Desired.ControlPlane.InfrastructureMachineTemplate.GetKind())
+			return objError(err, "failed to update", s.Desired.ControlPlane.InfrastructureMachineTemplate)
 		}
 
 		// The controlPlaneObject.Spec.machineTemplate.infrastructureRef has to be updated in the desired object
 		err = contract.ControlPlane().InfrastructureMachineTemplate().Set(s.Desired.ControlPlane.Object, refToUnstructured(cpInfraRef))
 		if err != nil {
-			return kerrors.NewAggregate([]error{errors.Wrapf(err, "failed to update the %s object", s.Desired.ControlPlane.Object.GetKind()), cleanup()})
+			return kerrors.NewAggregate([]error{objError(err, "failed to update the %s object", s.Desired.ControlPlane.Object), cleanup()})
```

Comment:
This seems harder to follow to me than the original version - at least at first glance. With errors.Wrapf it's clear what's going in and out.